apps/web/src/app/api/commits/[commitUuid]/documents/route.ts

```diff
@@ -0,0 +1,45 @@
+import materializeDocumentsAtCommit from '$/services/commitService/materializeDocumentsAtCommit'
+import { DocumentType } from '$core/schema'
+import createDocumentVersion from '$core/services/documentVersions/create'
+import { NextRequest, NextResponse } from 'next/server'
+import { z } from 'zod'
+
+export async function GET(
+  req: NextRequest,
+  { commitUuid }: { commitUuid: string },
+) {
+  try {
+    const staged = Boolean(req.nextUrl.searchParams.get('staged') || false)
+    const nodes = await materializeDocumentsAtCommit(commitUuid, staged)
+
+    return NextResponse.json(nodes)
+  } catch (err) {
+    return NextResponse.json(
+      { error: err.message },
+      { status: err.statusCode || 500 },
+    )
+  }
+}
+
+const postParams = z.object({
+  parentId: z.number().optional(),
+  documentType: z.enum([DocumentType.Folder, DocumentType.Document]).optional(),
+  name: z.string(),
+})
+
+export async function POST(
+  req: NextRequest,
+  { commitUuid }: { commitUuid: string },
+) {
+  try {
+    const data = postParams.parse(await req.json())
+    const docVersion = await createDocumentVersion({ ...data, commitUuid })
+
+    return NextResponse.json(docVersion, { status: 201 })
+  } catch (err) {
+    return NextResponse.json(
+      { error: err.message },
+      { status: err.statusCode || 500 },
+    )
+  }
+}
```

Comment:
Please let's try to use server actions unless there is a good reason to not use it. Just to be consistent

Reply:
cannot make it work for some reason I get this error

![image](https://github.com/user-attachments/assets/134354a3-a4ff-4951-8b3f-e477da377e92)

Reply:
Take a look into setup or login